astroquery/vo/core.py

```diff
@@ -0,0 +1,237 @@
+            Legal values are listed below.
+        source : string, optional
+            Any substring in ivoid (a unique identifier of the service)
+        publisher : string, optional
+            The name of any publishing organization (e.g., "stsci", "heasarc")
+        order_by : string, optional
+            What column to order it by.  The returned columns are:
+                "waveband","short_name","ivoid","res_description","access_url","reference_url","publisher", service_type"
+        logic_string : string, optional
+            Any other string you want to add to the ADQL where clause, should start with " and ".
+        verbose : bool, optional
+            Default is False.
+            When True, the ADQL query computed from the keyword arguments will be printed.
+        """
+
+        adql = self._build_adql(**kwargs)
+        if adql is None:
+            raise ValueError('Unable to compute query based on input arguments.')
+
+        if kwargs.get('verbose'):
+            print('Registry:  sending query ADQL = {}\n'.format(adql))
+
+        url = self._REGISTRY_TAP_SYNC_URL
+
+        tap_params = {
+            "request": "doQuery",
+            "lang": "ADQL",
+            "query": adql
+        }
+
+        response = utils.try_query(url, post_data=tap_params, timeout=self._TIMEOUT, retries=self._RETRIES)
+
+        if kwargs.get('verbose'):
+            print('Queried: {}\n'.format(response.url))
+
+        aptable = utils.astropy_table_from_votable_response(response)
+        return aptable
+
+    def _build_adql(self, **kwargs):
+
+        # Default values
+        service_type = ""
+        keyword = ""
+        waveband = ""
+        source = ""
+        publisher = ""
+        order_by = ""
+        logic_string = " and "
+
+        # Find the keywords we recognize
+        for key, val in kwargs.items():
+            if key == 'service_type':
+                service_type = val
+            elif key == 'keyword':
+                keyword = val
+            elif key == 'waveband':
+                waveband = val
+            elif key == 'source':
+                source = val
+            elif key == 'publisher':
+                publisher = val
+            elif key == 'order_by':
+                order_by = val
+            elif key == 'logic_string':
+                logic_string = val
+
+        ##
+        if "image" in service_type.lower():
+            service_type = "simpleimageaccess"
+        elif "spectr" in service_type.lower():
+            service_type = "simplespectralaccess"
+        elif "cone" in service_type.lower():
+            service_type = "conesearch"
+        elif 'tap' in service_type or 'table' in service_type:
+            service_type = "tableaccess"
+        else:
+            print("ERROR: please give a service_type that is one of image, spectral, cone, or table")
+            return None
```

Comment:
you should raise an error instead, i.e.: `raise ValueError("service_type must be one of...`

Reply:
Addressed